clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1287,6 +1287,16 @@ public void testCommitAsyncNegativeOffset() {
         assertTrue(mockOffsetCommitCallback.exception instanceof IllegalArgumentException);
     }
 
+    @Test(expected = CommitFailedException.class)
+    public void testCommitOffsetSyncCallbackWithoutNonRetriableException() {
```

Comment:
@ijuma , it took me a while to absorb the reason behind the suggested changes (Please consider that I am a newbie to the related classes). Although I made those changes, still my test has some glitches as it forever remains in the loop of `ConsumerNetworkClient.poll`. Can I have your pointers to fix that, please ?